refnx/analysis/parameter.py

```diff
@@ -231,6 +231,13 @@ def __init__(self):
         self.stderr = None
         self.chain = None
 
+    def __hash__(self):
```

Comment:
Does it have to be this complicated? Can't you just use the id(self)?
(It's not clear to me why there needs to be a hash, but I'll get to that later).

Reply:
as far as I can tell not really. I was coming up against a TypeError last night of 'unhashable type: Parameter'. The only way I could see to deal with it was that. Although it may be that this hash function is overly complex (I was aiming to be thorough).